src/sentry/integrations/utils/code_mapping.py

```diff
@@ -117,13 +137,18 @@ def _straight_path_logic(self, frame_file_path: str) -> None:
         # - some/path/foo.tsx
         # - ./some/path/foo.tsx
         # - /some/path/foo.js
+        # - app:///some/path/foo.js
+        # - ../../../some/path/foo.js
+        # - app:///../some/path/foo.js
 
-        start_at_index = 2 if frame_file_path.startswith("./") else 0
+        start_at_index = get_straight_path_prefix_end_index(frame_file_path)
         backslash_index = frame_file_path.find("/", start_at_index)
-        self.root = frame_file_path[0:backslash_index]  # some or .some
         dir_path, self.file_name = frame_file_path.rsplit("/", 1)  # foo.tsx (both)
+        self.root = frame_file_path[0:backslash_index]  # some or .some
         self.dir_path = dir_path.replace(self.root, "")  # some/path/ (both)
-        self.file_and_dir_path = frame_file_path.replace("./", "")  # some/path/foo.tsx (both)
+        self.file_and_dir_path = remove_straight_path_prefix(
```

Comment:
Try to use this instead, it will get rid of the new functions:

```
file_and_dir_path = frame_file_path  # I can't remember; do we need deep copy?
for _rep in ["app:///", "../", "./"]:
     file_and_dir_path = file_and_dir_path.replace(_rep, "")
self.file_and_dir_path = file_and_dir_path
```

Reply:
This function is used in 3 different spots, which is why I made the function. Should I replace them all with this?

Reply:
Keep `remove_straight_path_prefix` and put the logic in there.

It is my fault this code is so clunky. I can see how I could have written it better.

Either way, we will fix all of this when we refactor this code.

Thanks for your perseverance!